packages/background/src/infrastructure/stores/migrator/migrations/migration-47.ts

```diff
@@ -0,0 +1,92 @@
+
+/**
+ * This migration adds the zkSync alpha testnet network to the networks
+ */
+export default {
+    migrate: async (persistedState: BlankAppState) => {
+        const { availableNetworks } = persistedState.NetworkController;
+        const updatedNetworks = { ...availableNetworks };
+
+        updatedNetworks.SCROLL_L1_TESTNET = {
+            name: 'scroll_l1_testnet',
+            desc: 'Scroll L1 Testnet',
+            chainId: 534351,
+            networkVersion: '534351',
+            nativeCurrency: {
+                name: 'Ether',
+                symbol: 'TSETH',
+                decimals: 18,
+            },
+            iconUrls: [
+                'https://raw.githubusercontent.com/block-wallet/assets/master/blockchains/scroll/info/logo.png',
+            ],
+            isCustomNetwork: true,
+            enable: true,
+            test: true,
+            order: 8,
+            features: [FEATURES.SENDS],
+            ens: false,
+            showGasLevels: true,
+            rpcUrls: [`https://prealpha.scroll.io/l1`],
+            blockExplorerUrls: ['https://l1scan.scroll.io/'],
+            blockExplorerName: 'Scroll L1 Explorer',
+            actionsTimeIntervals: { ...TESTNET_TIME_INTERVALS_DEFAULT_VALUES },
+            tornadoIntervals: {
+                depositConfirmations: DEFAULT_TORNADO_CONFIRMATION,
+                derivationsForward: DERIVATIONS_FORWARD,
+            },
+            nativelySupported: true,
+        };
+        updatedNetworks.SCROLL_L2_TESTNET = {
+            name: 'scroll_l2_testnet',
+            desc: 'Scroll L2 Testnet',
+            chainId: 534354,
+            networkVersion: '534354',
+            nativeCurrency: {
+                name: 'Ether',
+                symbol: 'TSETH',
+                decimals: 18,
+            },
+            iconUrls: [
+                'https://raw.githubusercontent.com/block-wallet/assets/master/blockchains/scroll/info/logo.png',
+            ],
+            isCustomNetwork: true,
+            enable: true,
+            test: true,
+            order: 9,
+            features: [FEATURES.SENDS],
+            ens: false,
+            showGasLevels: true,
+            rpcUrls: [`https://prealpha.scroll.io/l2`],
+            blockExplorerUrls: ['https://l2scan.scroll.io/'],
+            blockExplorerName: 'Scroll L2 Explorer',
+            actionsTimeIntervals: { ...TESTNET_TIME_INTERVALS_DEFAULT_VALUES },
+            tornadoIntervals: {
+                depositConfirmations: DEFAULT_TORNADO_CONFIRMATION,
+                derivationsForward: DERIVATIONS_FORWARD,
+            },
+            nativelySupported: true,
+        };
+
+        updatedNetworks.LOCALHOST = {
+            ...updatedNetworks.LOCALHOST,
+            order: 10,
+        };
+        return {
+            ...persistedState,
+            NetworkController: {
+                ...persistedState.NetworkController,
+                availableNetworks: { ...updatedNetworks },
```

Comment:
Shouldn't we invoke `normalizeNetworksOrder` here? @antonyayoub

Reply:
this was already applied in the [migration](https://github.com/block-wallet/extension/blob/master/packages/background/src/infrastructure/stores/migrator/migrations/migration-45.ts) for that PR, the order is up to date and grouped by `testnet` field in https://github.com/block-wallet/extension/blob/cdc7914524a0f0b9b91e6c9c8a66eef6e69705ba/packages/background/src/utils/constants/networks.ts#L135 so it shouldn't be necessary as Lea added them with the correct order

Reply:
@julianariel @fpezzola, I believe we should use normalizeNetworksOrder here or **at least add the new network's order property dynamically as the last of order for testnet or mainnet**. Because what if the user added more testnets than the currently assumed order number by default(8), and then the extension got updated?  Networks might be overlapping in the Order of properties. ( it wouldn't cause errors, though it will be normalised as soon as the user tries to order the networks,  add a new network or edit a network. )